File: Практическая работа №3/src/main/java/io/github/ikbo0619/testing/team6/durak/Hand.java
package io.github.ikbo0619.testing.team6.durak;

import java.util.ArrayList;

public class Hand {
    private final ArrayList<Card> cards;

    public Hand() {
        cards = new ArrayList<>();
    }

    public Hand(int n) {
        cards = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            Card thisCard = new Card();
            cards.add(thisCard);
        }
    }

    public void add(Card c) {
        cards.add(c);
    }

    public void remove(Card c) {
        cards.remove(c);
    }

    public int size() {
        return cards.size();
    }

    public boolean needsToDraw() {
        return size() > 6;
    }

    public int numberToDraw() {
        return needsToDraw() ? 0 : 6 - size();
    }

    public ArrayList<Card> getCards() {
        return cards;
    }

    public Card getCardByIndex(int i) {
        return cards.get(i);
    }

    public Card useCardByIndex(int i) {
        return cards.remove(i);
    }

    @Override
    public String toString() {
        String ret = "";
        for (Card c : cards)
            ret += c + "\n";
        return ret;
    }
}

File: Практическая работа №6/№3 Имеющаяся программная система/src/main/kotlin/Hand.kt
class Hand {
    var cards = ArrayList<Card>()
        private set

    fun add(card: Card) = cards.add(card)

    fun size(): Int = cards.size

    fun numberToDraw() =
        if (size() > 6) 0
        else 6 - size()

    fun getCardByIndex(index: Int) = cards[index]

    fun useCardByIndex(index: Int) = cards.removeAt(index)

    override fun toString() = buildString {
        for (c in cards) append("$c\n")
    }
}
